### ISSUE TITLE
Propagate exception context from service layer

### ISSUE THREAD

**Issue by sshah-wework** [MarquezProject/marquez#426]
Currently we're swallowing the context of an exception when it's thrown from the service layer. For example:

```
      String err = "error updating job run state";
      log.error(err, e);
      throw new MarquezServiceException();
```

When we're logging exceptions, we'll want to make sure to wrap the original exception and include the error summary string:

```
      String err = "error updating job run state";
      log.error(err, e);
      throw new MarquezServiceException(err, e);
```

**Comment by sshah-wework** [comment]
Actually, if we do our exception logging from `log.error`, the context is getting passed correctly. so this is probably ok.

**Comment by sshah-wework** [comment]
Hm, there are some parts of code where the context still gets dropped, so re-opening. for example:

```
    } catch (UnableToExecuteStatementException e) {
      log.error(e.getMessage());
      throw new MarquezServiceException();
    }
```

should be:

```
    } catch (UnableToExecuteStatementException e) {
      log.error(e.getMessage(), e);
      throw new MarquezServiceException();
    }
```

Ideally instead of using `e.getMessage()` which has a low-level error message, we should put a higher-level message (e.g. "failed to fetch job")

**Comment by wslulciuc** [comment]
Fixed #627
